src/Gui/GLForwardDecls.h

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+/***************************************************************************
+ *   Copyright (c) 2025 FreeCAD Project                                    *
+ *                                                                         *
+ *   This file is part of FreeCAD.                                         *
+ *                                                                         *
+ *   FreeCAD is free software: you can redistribute it and/or modify it    *
+ *   under the terms of the GNU Lesser General Public License as           *
+ *   published by the Free Software Foundation, either version 2.1 of the  *
+ *   License, or (at your option) any later version.                       *
+ *                                                                         *
+ *   FreeCAD is distributed in the hope that it will be useful, but        *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of            *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU      *
+ *   Lesser General Public License for more details.                       *
+ *                                                                         *
+ *   You should have received a copy of the GNU Lesser General Public      *
+ *   License along with FreeCAD. If not, see                               *
+ *   <https://www.gnu.org/licenses/>.                                      *
+ *                                                                         *
+ **************************************************************************/
+
+//! GLForwardDecls.h
+//! provides OpenGL forward declarations
+
+#ifndef GUI_GLFORWARDDECLS_H
+#define GUI_GLFORWARDDECLS_H
+
+// Basic OpenGL types
+using GLenum     = unsigned int;    //!< Unsigned integer used for enumerations.
+using GLboolean  = unsigned char;   //!< Boolean value (usually 0 or 1).
+using GLbitfield = unsigned int;    //!< Bitfield type.
+using GLvoid     = void;            //!< Void type, for pointers.
+using GLbyte     = signed char;     //!< Signed byte.
+using GLubyte    = unsigned char;   //!< Unsigned byte.
+using GLshort    = short;           //!< Signed short integer.
+using GLushort   = unsigned short;  //!< Unsigned short integer.
+using GLint      = int;             //!< Signed integer.
+using GLuint     = unsigned int;    //!< Unsigned integer.
+using GLsizei    = int;             //!< Size type (integer).
+using GLfloat    = float;           //!< Floating point value.
+using GLclampf   = float;           //!< Floating point value clamped between 0 and 1.
+using GLdouble   = double;          //!< Double precision floating point value.
+using GLclampd   = double;          //!< Double precision clamped value.
```

Comment:
As far as I know, [GL types are fixed in size](https://www.khronos.org/opengl/wiki/OpenGL_Type), so we need to use `uint32_t` and friends to ensure portability.

Can you add size checks for these data types using `static_assert`, preferably with checking against gl in a new cpp file?

Reply:
Another way to reduce build time could be to avoid using opengl types in header files, would that be enough maybe?

Reply:
By checking the files on your system, you only see how it looks for you, at this time.
On macOS `gl.h` includes `gltypes.h`, which looks like this:
```c++
typedef uint32_t GLbitfield;
typedef uint8_t  GLboolean;
typedef int8_t   GLbyte;
typedef float    GLclampf;
typedef uint32_t GLenum;
typedef float    GLfloat;
typedef int32_t  GLint;
typedef int16_t  GLshort;
typedef int32_t  GLsizei;
typedef uint8_t  GLubyte;
typedef uint32_t GLuint;
typedef uint16_t GLushort;
typedef void     GLvoid;
```

By checking with static assert, we don't have to assume anything - maybe I'm unlucky but it usually backfires when I assume stuff about the size.

Reply:
I'd prefer it, but it's not a hill i'm willing to die on. @kadet1090, what do you think?

Reply:
Hmm, I really don't like idea of making "forward declarations" for OpenGL. Type sizes can be implementation dependent (even if they should not) and this is not forward declaration but copying and pasting actual declarations of the types.